src/pybind/mgr/cephadm/module.py

```diff
@@ -1792,44 +1831,19 @@ def update_mgrs(self, spec, services):
             # otherwise, remove *any* mgr
             if num_to_remove > 0:
                 for d in services:
-                    to_remove_damons.append(('%s.%s' % (d.service_type, d.service_instance), d.nodename))
-                    num_to_remove -= 1
-                    if num_to_remove == 0:
-                        break
+                    if d.service_instance != self.get_mgr_id():
+                        to_remove_damons.append(('%s.%s' % (d.service_type, d.service_instance), d.nodename))
+                        num_to_remove -= 1
+                        if num_to_remove == 0:
+                            break
             c = self._remove_daemon(to_remove_damons)
             c.add_progress('Removing MGRs', self)
             c.update_progress = True
             return c
 
         else:
-            # we assume explicit placement by which there are the same number of
-            # hosts specified as the size of increase in number of daemons.
-            num_new_mgrs = spec.count - num_mgrs
-            if len(spec.placement.hosts) < num_new_mgrs:
-                raise RuntimeError(
-                    "Error: {} hosts provided, expected {}".format(
-                        len(spec.placement.hosts), num_new_mgrs))
-
-            for host_spec in spec.placement.hosts:
-                if host_spec.name and len([d for d in services if d.service_instance == host_spec.name]):
-                    raise RuntimeError('name %s alrady exists', host_spec.name)
-
-            for host_spec in spec.placement.hosts:
-                if host_spec.name and len([d for d in services if d.service_instance == host_spec.name]):
-                    raise RuntimeError('name %s alrady exists', host_spec.name)
-
-            self.log.info("creating {} managers on hosts: '{}'".format(
-                num_new_mgrs, ",".join([_spec.hostname for _spec in spec.placement.hosts])))
-
-            args = []
-            for host_spec in spec.placement.hosts:
-                host = host_spec.hostname
-                name = host_spec.name or self.get_unique_name(host, services)
-                args.append((host, name))
-            c = self._create_mgr(args)
-            c.add_progress('Creating MGRs', self)
```

Comment:
the progress reference was dropped

Reply:
Yeah.. I think we should add it in the generic helper, but I thought we'd do that in a separate PR.

Reply:
i was going to come back to this since it's missing from all of the others